File: 명관/20230613/로또 6603.cpp
#include <iostream>
#include <vector>
using namespace std;

int S[13];
int lotto[6];
int k;
//start는 S의 인덱스, depth는 lotto의 인덱스 
void getCombi(int start, int depth) {
	if (depth == 6) {
		for (int i = 0; i < 6; i++) {
			cout << lotto[i] << " ";
		}
		cout << endl;
		return;
	}
	//start부터 k-1 까지 순서
	for (int i = start; i < k; i++) {
		lotto[depth] = S[i];
		getCombi(i+1, depth+1);
	}
}
int main(void) {
	while (1) {
		cin >> k;
		if (k == 0) break;
		for (int i = 0; i < k; i++) {
			cin >> S[i];
		}
		getCombi(0, 0);
		cout << endl;
	}
}

File: 명관/20230613/로또 6603.java
import java.io.*;
import java.util.*;
 
public class Main {
    static int N ;
    static int[] arr;
    static boolean[] result;
 
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
 
        while (true){
            StringTokenizer st = new StringTokenizer(br.readLine());
            N = Integer.parseInt(st.nextToken());
 
            if(N==0){
                break;
            }
            arr = new int[N];
            result = new boolean[N];
            for (int i = 0; i <N ; i++) {
                arr[i] = Integer.parseInt(st.nextToken());
            }
 
            DFS(0, 0);
            System.out.println();
 
        }
    }
    private static void DFS(int start, int depth){
        if(depth == 6){
            for (int i = 0; i <N ; i++) {
                if(result[i]){
                    System.out.print(arr[i] + " ");
                }
            }
            System.out.println();
        }
 
        for (int i = start; i <N ; i++) {
            result[i] = true;
            DFS(i+1, depth+1);
            result[i] = false;
        }
 
    }
}